pkg/pullsync/pullsync.go

```diff
@@ -179,7 +181,8 @@ func (s *Syncer) SyncInterval(ctx context.Context, peer swarm.Address, bin uint8
 			s.logger.Debug("storage has", "error", err)
 			continue
 		}
-		if !have {
+		po := swarm.Proximity(a.Bytes(), peer.Bytes())
```

Comment:
as an optimization, let's move the po check before the storage.Has to save on disk usage